deps/key-value-storage/src/postgres/mod.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2025 by Alibaba.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! PostgreSQL backend for the key-value storage.
+
+use std::env;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Context};
+use async_trait::async_trait;
+use derivative::Derivative;
+use serde::Deserialize;
+use sqlx::PgPool;
+use sqlx::{postgres::PgPoolOptions, query, Row};
+use tracing::{debug, info, instrument};
+
+use crate::{KeyValueStorage, KeyValueStorageError, Result, SetContext};
+
+/// The maximum number of connections to the PostgreSQL database.
+pub const MAX_CONNECTIONS: u32 = 5;
+
+/// The name of the key column.
+pub const KEY_COLUMN: &str = "key";
+
+/// The name of the value column.
+pub const VALUE_COLUMN: &str = "value";
+
+/// The environment variable name for the PostgreSQL URL.
+pub const POSTGRES_URL_ENV_VAR: &str = "POSTGRES_URL";
+
+#[derive(Deserialize, Derivative, Clone, PartialEq)]
+#[derivative(Debug)]
+pub struct Config {
+    /// The name of the PostgreSQL database.
+    #[serde(default)]
```

Comment:
nit: you can move this to the struct I think.

Reply:
moved.